src/thermal_conductivity/scattering.f90

```diff
@@ -94,10 +94,15 @@ subroutine generate(sr, qp, dr, uc, fct, fcf, opts, tmr, mw, mem)
             call lo_stop_gracefully(['This routine only works with FFT meshes'], lo_exitcode_param, __FILE__, __LINE__)
         end select
 
-        if (opts%seed .gt. 0) then
-            rseed = 1.0 / real(opts%seed, r8)
+        if (opts%seed .ge. 0) then
+            if (opts%seed == 0) then
+                rseed = 0.0_r8
+            else
+                rseed = 1.0 / real(opts%seed, r8)
+            end if
```

Comment:
```suggestion
        if (opts%seed .ge. 0) then
            rseed = 1.0 / real(opts%seed, r8)
```

extra check for 0 not needed anymore

Reply:
Wouldn't this create problem if the seed is actually 0 since we divide by `opts%seed` ?

Reply:
ah right. I confused `.ge.` with `>`, but that's `.gt.`. To correct my suggestion: Let us just set the seed when `seed .gt. 0`, see also the comment in the options (positive integer...)